sdk/core/core-client/src/serviceClient.ts

```diff
@@ -88,7 +97,11 @@ export class ServiceClient {
    */
   constructor(options: ServiceClientOptions = {}) {
     this._requestContentType = options.requestContentType;
-    this._baseUri = options.baseUri;
+    this._endpoint = options.endpoint ?? options.baseUri;
+    if (options.baseUri) {
+      const logger = createClientLogger("ServiceClientOptions");
+      logger.warning("This baseUri option has been deprecated, please use endpoint instead!");
```

Comment:
@joheredi @xirzec  I add the warning here. what do you think ?

Reply:
Minor rewording

```suggestion
      logger.warning("The baseUri option for SDK Clients has been deprecated, please use endpoint instead.");
```